hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/FileTransferHandlerVaadinUpload.java

```diff
@@ -131,26 +132,23 @@ public OutputStream receiveUpload(final String fileName, final String mimeType)
 
         // we return the outputstream so we cannot close it here
         @SuppressWarnings("squid:S2095")
-        OutputStream outputStream = ByteStreams.nullOutputStream();
+        final PipedOutputStream outputStream = new PipedOutputStream();
+        Optional<PipedInputStream> inputStream = Optional.empty();
         try {
-            outputStream = createOutputStreamForTempFile();
-            this.mimeType = mimeType;
-            publishUploadProgressEvent(fileUploadId, 0, 0, getTempFilePath());
-
+            inputStream = Optional.of(new PipedInputStream(outputStream));
+            publishUploadProgressEvent(fileUploadId, 0, 0);
+            startTransferToRepositoryThread(inputStream.orElseThrow(IOException::new), fileUploadId, mimeType);
         } catch (final IOException e) {
-            LOG.error("Creating temp file for upload failed {}.", e);
-            if (outputStream != null) {
-                try {
-                    outputStream.close();
-                } catch (final IOException e1) {
-                    LOG.error("Closing output stream caused an exception {}", e1);
-                }
-            }
-
+            LOG.error("Creating piped Stream failed {}.", e);
             setFailureReasonUploadFailed();
             setUploadInterrupted();
+            tryToCloseIOStream(outputStream);
```

Comment:
Check if the streams can be closed before calling setFailureReasonUploadFailed and setUploadInterrupted.